expected/poly.out

```diff
@@ -403,13 +403,17 @@ LINE 1: SELECT spoly '{(0d,0d),(10d,10d),(0d,10d),(10d,0d)}';
                      ^
 --- degenerate polygons -----
 SELECT spoly '{(0d,1d),(0d,2d),(0d,3d)}';
-ERROR:  spherepoly_from_array: a line segment overlaps or polygon too large
-LINE 1: SELECT spoly '{(0d,1d),(0d,2d),(0d,3d)}';
-                     ^
+              spoly              
+---------------------------------
+ {(0d , 1d),(0d , 2d),(0d , 3d)}
+(1 row)
+
```

Comment:
This doesn't seem correct. This spoly and the following one _should_ fail, no? They do not seem to be valid polygons to me.

Reply:
@esabol I rolled back the changes in PR #36 because it broke the creation of valid polygons. I looked at the issue where we decided to consider degenerate polygons as invalid, but I can't remember why we decided to so, unfortunately. Probably, due to OGC specification of valid polygons. This bug actually makes pgsphere unusable in production, but invalidation of degenerate polygons don't produce any problems. Despite of OGC specification of valid polygons, in practice, we may consider support of genenerate polygons as well to make pgsphere more usable.

Reply:
@esabol I understand the problem with sline_sline_pos. I'm going to propose another solution but it takes more time to implement and test. The core reason is in using of spoint_at_sline in sline_sline_pos before checking for connectivity of adjacent segments. In past, the connectivity check was prior to the calling of spoint_at_sline.

I believe it is reasonable to apply this PR and increment patch number, because the big is severe. But next commit we improve sline_sline_pos. It more risky and needs more testing.